### DISCUSSION TITLE
Multiple connections between elements

### DISCUSSION THREAD

**Discussion by mracsko** [likec4/likec4#983] [Q&A]
I try to model some of the behavior between two systems, and I run into a limitation, or I am doing something wrong.

In some cases, I can have multiple relationships between two elements:
- There are multiple `relationship` defined, modeling different abstraction levels (data flow, actual communication (rest, messaging), etc.), and two different view can have different actual connections.
- There are multiple connections with the same `relationship` type for modelling different communication, like a request that retrieves some data and another that is handling a completely different data. Or one Rest and one gRPC call.

When I tried to model this, it always picking the last connection in the list, ignoring the others. This is even the case when I have a different relationship type. I cannot even replicate the behavior explained in #218 or #663 with the `[..]`. **Please see the example below.**

My two main questions are:
- Am I trying to do something that is not possible currently, or am I just doing it wrong and there is a way to model multiple connections?
- My second answer is only relevant if the multiple connections can be achieved. Can you name a connection so you can reference it later? For example you have 2 connections between two systems, and you want to exclude one of those and not the other in a view. Tags can work I assume, but a specific name based reference could be less cumbersome.

**Example:**
```
specification {
  element system
  
  relationship data-flow
  relationship messaging
  relationship rest
}

model {
  system1 = system 'System 1'
  system2 = system 'System 2'
  
  system1 -[data-flow]-> system2 'data requered for System 2 to do its job'
  system1 -[messaging]-> system2 'urgent notifications'
  system1 -[rest]-> system2 'sending hourly update by calling rest endpoint of System 2'
  system1 -[rest]-> system2 'sending daily summary data by calling rest endpoint of System 2'
  system2 -[rest]-> system1 'polling System 1 to get latest updates'
}

views {

  view index {
    title 'Multiple connections example'

    include *
  }

}
```

**Comment by davydkov** [comment]
Hey @mracsko 

> When I tried to model this, it always picking the last connection in the list, ignoring the others.
> ...
> Am I trying to do something that is not possible currently, or am I just doing it wrong 

No, your expectations absolutely right. There was an issue, fixed in #991 

Your example looks like:

![image](https://github.com/user-attachments/assets/d11da5e5-b63f-4a11-b678-24b3aacafc37)

Do i get you right, this is what you expected?


And second question:

> Can you name a connection so you can reference it later?

Not yet implemented at the moment. For now i would suggest use tags and filter:

```
  exclude
     source -> target
     where tag is #sometag
```
